c7n/tags.py

```diff
@@ -671,3 +660,154 @@ def process(self, resources):
                         "Exception renaming tag set \n %s" % (
                             f.exception()))
         return resources
+
+
+class UniversalTag(Tag):
+    """Applies one or more tags to the specified resources.
+    """
+
+    batch_size = 20
+    permissions = ('resourcegroupstaggingapi:TagResources',)
+
+    def process(self, resources):
+        self.id_key = self.manager.get_model().id
+
+        # Legacy
+        msg = self.data.get('msg')
+        msg = self.data.get('value') or msg
+
+        tag = self.data.get('tag', DEFAULT_TAG)
+        tag = self.data.get('key') or tag
+
+        # Support setting multiple tags in a single go with a mapping
+        tags = self.data.get('tags', {})
+
+        if msg:
+            tags[tag] = msg
+
+        batch_size = self.data.get('batch_size', self.batch_size)
+
+        _common_tag_processer(
+            self.executor_factory, batch_size, self.concurrency,
+            self.process_resource_set, self.id_key, resources, tags, self.log)
+
+    def process_resource_set(self, resource_set, tags):
+        client = utils.local_session(
+            self.manager.session_factory).client('resourcegroupstaggingapi')
+
+        arns = []
+        for r in resource_set:
+            _id = r[self.id_key]
+            if 'arn' in _id[:3]:
```

Comment:
table for now, but should probably get an arn class going rather than adhoc arn parsing.